presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsTableHandle.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tpcds;
+
+import com.facebook.presto.spi.ConnectorTableHandle;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+
+public class TpcdsTableHandle
+        implements ConnectorTableHandle
+{
+    private final String tableName;
+    private final double scaleFactor;
+
+    @JsonCreator
+    public TpcdsTableHandle(@JsonProperty("tableName") String tableName, @JsonProperty("scaleFactor") double scaleFactor)
+    {
+        this.tableName = requireNonNull(tableName, "tableName is null");
+        this.scaleFactor = requireNonNull(scaleFactor, "scaleFactor is null");
```

Comment:
`requireNonNull` not needed.
`checkArgument(scaleFactor > 0)` could be useful instead.